docs/walkthrough.md

```diff
@@ -0,0 +1,316 @@
+         38   152    EQ(17, 17, 0)        13 45 10 00
+         39   156    JNZI(17, 41)         73 44 00 29   conditionally jump to byte 164
+         40   160    JI(43)               90 00 00 2b   jump to byte 172
+         41   164    LOG(16, 0, 0, 0)     33 40 00 00
+         42   168    JI(49)               90 00 00 31   jump to byte 196
+         43   172    LW(19, 63, 4)        5d 4f f0 04
+         44   176    ADDI(17, 20, 16)     50 45 40 10
+         45   180    SW(20, 19, 2)        5f 51 30 02
+         46   184    ADDI(17, 20, 16)     50 45 40 10
+         47   188    LW(17, 20, 2)        5d 45 40 02
+         48   192    LOGD(0, 0, 16, 17)   34 00 04 11
+         49   196    LW(16, 63, 1)        5d 43 f0 01
+         50   200    RET(0)               24 00 00 00
+         51   204    Undefined            00 00 00 00
+         52   208    Undefined            00 00 00 01
+         53   212    Undefined            00 00 00 00
+         54   216    Undefined            00 00 00 00
+         55   220    Undefined            00 00 00 00
+         56   224    Undefined            00 00 00 05
+         57   228    Undefined            00 00 00 00
+         58   232    Undefined            00 00 00 00
+         59   236    Undefined            00 00 00 00
+         60   240    Undefined            00 00 00 08
+```
+
+We can recognize the `while` loop by the conditional jumps `JNZI(..)`. The condition just before the first jump can be identified by `LT(..)` instruction (for `<`). Some notable instructions that are generated only once in our code include `MUL(..)` for multiplication and `LOG(.., 0, 0, 0)` from the `log` function.
+
+## Setting up the debugging
+
+We can start up the debug infrastructure. On a new terminal session run `fuel-core --db-type in-memory`; we need to have that running because it actually executes the program. Now we can fire up the debugger itself: `fuel-debugger`. Now
+if everything is set up correctly, you shoould see the debugger prompt (`>> `). You can use `help` command to list available commands.
+
+Now we would like to inspect the program while it's running. To do this, we first need to send the script to the executor, i.e. `fuel-core`. To do so, we need a *transaction specification*, `tx.json`. It looks something like this:
+
+```json
+{
+    "Script": {
+        "byte_price": 0,
+        "gas_price": 0,
+        "gas_limit": 1000000,
+        "maturity": 0,
+        "script": [],
+        "script_data": [],
+        "inputs": [],
+        "outputs": [],
+        "witnesses": [],
+        "receipts_root": "0000000000000000000000000000000000000000000000000000000000000000"
+    }
+}
+```
+
+However, the key `script` in should contain the actual bytecode to execute, i.e. the contents of `out/debug/dbg_example.bin` as a JSON array. The following command can be used to generate it:
+
+```bash
+python -c 'print(list(open("out/debug/dbg_example.bin", "rb").read()))'
```

Comment:
hmm this didn't quite work for me, I got this escaped version:

```
python -c 'print(list(open("out/debug/dbg_example.bin", "rb").read()))'
['\x90', '\x00', '\x00', '\x04', 'G', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\xcc', ']', '\xfc', '\xc0', '\x01', '\x10', '\xff', '\xf3', '\x00', '\x1a', 'L', 'P', '\x00', '\x91', '\x00', '\x00', '\x18', 'P', 'A', '0', '\x08', ']', 'C', '\xf0', '\x00', '_', 'M', '\x00', '\x01', 'P', 'A', '0', '\x00', ']', 'C', '\xf0', '\x01', '_', 'M', '\x00', '\x00', 'P', 'A', '0', '\x00', ']', 'E', '0', '\x00', ']', 'C', '\xf0', '\x02', '\x16', 'A', '\x14', '\x00', 's', '@', '\x00', '\x14', '\x90', '\x00', '\x00', '"', 'P', 'A', '0', '\x00', 'P', 'A', '0', '\x00', ']', 'E', '0', '\x00', ']', 'C', '\xf0', '\x00', '\x10', 'A', '\x04', '@', '_', 'M', '\x00', '\x00', 'P', 'A', '0', '\x08', 'P', 'A', '0', '\x08', ']', 'E', '0', '\x01', 'P', 'A', '0', '\x00', ']', 'A', '0', '\x00', '\x1b', 'A', '\x14', '\x00', '_', 'M', '\x00', '\x01', '\x90', '\x00', '\x00', '\x0e', 'P', 'A', '0', '\x08', ']', 'A', '0', '\x01', ']', 'G', '\xf0', '\x03', '\x13', 'E', '\x10', '\x00', 's', 'D', '\x00', '(', '\x90', '\x00', '\x00', '*', '3', '@', '\x00', '\x00', '\x90', '\x00', '\x00', '0', ']', 'K', '\xf0', '\x04', 'P', 'E', '0', '\x10', '_', 'M', ' ', '\x02', 'P', 'E', '0', '\x10', ']', 'E', '0', '\x02', '4', '\x00', '\x04', '\x11', ']', 'C', '\xf0', '\x01', '$', '\x00', '\x00', '\x00', 'G', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x01', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x05', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x08']
```

Reply:
Right, Python version 2/3 compatibility issues of course. I'll specify Python 3.